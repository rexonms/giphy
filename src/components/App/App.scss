@import './../../../node_modules/reset-css';

body {
  background: $black;
}
p {
  color: $white;
  font-family: $fontDefault;
  font-size: $fontSizeDefault;
  line-height: $fontSizeDefault * 1.3;
  color: $defaultFontColor;
}
.container {
  //width:100vw;
  //height: 100vh;
  margin: auto;
}