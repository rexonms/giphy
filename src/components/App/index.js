import React from 'react';
import { ConnectedRouter } from 'connected-react-router';

import { history } from '../../configuredStore';
import Routes from '../../Routes';
import Styles from './App.scss';
import packageJson from '../../../package.json';

const propTypes = {};
const defaultProps = {};
const App = () => (
  <div className={Styles.container} bundleversion={packageJson.version}>
    <ConnectedRouter history={history}>
      <Routes />
    </ConnectedRouter>
  </div>
);

App.defaultProps = defaultProps;
App.propTypes = propTypes;

export default App;
