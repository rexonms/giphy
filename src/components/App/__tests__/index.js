import React from 'react';
import { shallow } from 'enzyme';

import App from './../index';

describe('Welcome Component', () => {
  let props;
  let shallowApp;
  const app = () => {
    if (!shallowApp) {
      shallowApp = shallow(<App {...props} />);
    }
    return shallowApp;
  };
  beforeEach(() => {
    props = {};
    shallowApp = undefined;
  });

  it('should render previous snapshot', () => {
    expect(app()).toMatchSnapshot();
  });
  it('should have child components', () => {
    expect(app().find('Routes').exists()).toBe(true);
  });
});
