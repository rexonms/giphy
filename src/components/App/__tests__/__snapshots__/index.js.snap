// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome Component should render previous snapshot 1`] = `
<div
  bundleversion="0.1.0"
  className="container"
>
  <Connect(ConnectedRouter)
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Routes />
  </Connect(ConnectedRouter)>
</div>
`;
